plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/linearization/PureLinearizer.kt

```diff
@@ -26,16 +25,16 @@ class PureLinearizer(override val source: KtSourceElement?) : LinearizationConte
     override fun <R> withPosition(newSource: KtSourceElement, action: LinearizationContext.() -> R): R =
         PureLinearizer(newSource).action()
 
-    override fun newVar(type: TypeEmbedding): VariableEmbedding {
+    override fun freshAnonVar(type: TypeEmbedding): Exp.LocalVar {
         throw PureLinearizerError("newVar")
     }
 
     override fun inhaleForThisStatement(assumption: Exp) {
         throw PureLinearizerError("inhaleForThisStatement")
     }
 
-    override fun withNewScope(action: LinearizationContext.() -> Unit): Stmt.Seqn {
-        throw PureLinearizerError("withNewScope")
+    override fun asBlock(action: LinearizationContext.() -> Unit): Stmt.Seqn {
+        throw PureLinearizerError("withNewScopeToBlock")
```

Comment:
Nit: this is a general note (for more info, check [here](https://www.baeldung.com/java-errors-vs-exceptions#error)). In JVM programs, there is a distinction between `Error` and `Exception`. I would call the `PureLinearizerError` `PureLinearizerException`.